tests/cases/compiler/instantiationExpressionNoTypeParameterLeak1.ts

```diff
@@ -0,0 +1,24 @@
+// @strict: true
+// @noEmit: true
+
+// https://github.com/microsoft/TypeScript/issues/61041
+
+export const test1 = <X,>(g: <A>(x: X) => X) => g<string>;
+export const output1 = test1<number>((y: number) => 1);
+output1(1);
```

Comment:
The leaked type param was filtered out by the call to `isTypeParameterPossiblyReferenced` in `getObjectTypeInstantiation`. It was crucial for the bug that a body-less arrow function returned the instantiation expression. With a block function body `isTypeParameterPossiblyReferenced` would already return `true`.